Classes/Service/ContentService.php

```diff
@@ -222,7 +221,15 @@ public function moveRecord(array &$row, &$relativeTo, $parameters, DataHandler $
 				$row['tx_flux_column'] = $relativeToRecord['tx_flux_column'];
 				$row['colPos'] = $relativeToRecord['colPos'];
 				$row['sorting'] = $tceMain->resorting('tt_content', $relativeToRecord['pid'], 'sorting', abs($relativeTo));
+			} elseif (0 < (integer) $relativeTo) {
+				// moving to first position in colPos, means that $relativeTo is the pid of the containing page
+				$row['sorting'] = $tceMain->getSortNumber('tt_content', 0, $relativeTo);
+				// neither change fields tx_flux_column nor tx_flux_parent here!
+			} else {
+				$row['tx_flux_column'] = $row['tx_flux_parent'] = NULL;
```

Comment:
In future we try to avoid multiple assignments for better readable code.
Variable is always in first block.